### PR TITLE
[Bug] Obstruct et al no longer protect against status moves

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes the user will see?
[Burning Bulwark](https://dex.pokemonshowdown.com/moves/burningbulwark), [King's Shield](https://dex.pokemonshowdown.com/moves/kingsshield), [Obstruct](https://dex.pokemonshowdown.com/moves/obstruct) and [Silk Trap](https://dex.pokemonshowdown.com/moves/silktrap) no longer protect against status moves.

## Why am I making these changes?
These moves are not supposed to protect against status moves but currently they do.

## What are the changes from a developer perspective?
A new `BattlerTag` class `DamageProtectedTag` was created. `ContactStatStageChangeProtectedTag` and `ContactBurnProtectedTag` now inherit from that instead of `ProtectedTag`, and the protection check in `MoveEffectPhase` was updated to account for the change.

## How to test the changes?
`npm run test obstruct`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/test/moves/obstruct.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { Moves } from "#app/enums/moves";
+import { Stat } from "#app/enums/stat.js";
```
Comment (ben-lear) [inline src/test/moves/obstruct.test.ts:2]: nit: import .js

Comment (ben-lear) [inline src/test/moves/obstruct.test.ts:2]:
```suggestion
import { Stat } from "#app/enums/stat";
```

**Review by ben-lear — CHANGES_REQUESTED** [review]

File: src/test/moves/obstruct.test.ts (hunk 1 above)
Comment (DayKev) [reply src/test/moves/obstruct.test.ts:2]: :despair:

**Review by innerthunder — APPROVED** [review]
lgtm, though maybe a better name for the new tag would be `AttackMoveProtectedTag` (or just `AttackProtectedTag`)?

**Review by schmidtc1 — APPROVED** [review]

File: src/data/battler-tags.ts [hunk 2]
```diff
@@ -1180,6 +1180,8 @@ export class ProtectedTag extends BattlerTag {
   }
 }
 
+export class DamageProtectedTag extends ProtectedTag {}
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:1183]: Can we get some tsdoc on these new classes

File: src/data/battler-tags.ts [hunk 3]
```diff
@@ -1215,7 +1217,7 @@ export class ContactDamageProtectedTag extends ProtectedTag {
   }
 }
 
-export class ContactStatStageChangeProtectedTag extends ProtectedTag {
+export class ContactStatStageChangeProtectedTag extends DamageProtectedTag {
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:1220]: Each of these classes should have a tsdoc that explains what they're doing and which moves they're associated with

**Review by Tempo-anon — CHANGES_REQUESTED** [review]

**Review by Tempo-anon — APPROVED** [review]
Baneful bunker is somehow different. Gamefreak why.

**Pull request merged**